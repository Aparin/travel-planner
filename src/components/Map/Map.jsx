import React, { Component, createRef } from 'react';
import PropTypes from 'prop-types';
import ErrorBoundary from '../ErrorBoundary';

class Map extends Component {
  constructor(props) {
    super(props);
    this.map = createRef(null);
    this.center = props.center;
    this.onError = props.onError;
  }

  componentDidMount() {

  }

  render() {
    return (
      <ErrorBoundary>
        <div ref={this.map} style={{ width: '800px', height: '500px', display: 'inline-block' }} />
      </ErrorBoundary>
    );
  }
}

Map.propTypes = {
  center: PropTypes.arrayOf(PropTypes.number).isRequired,
  onError: PropTypes.func.isRequired,
};

export default Map;
