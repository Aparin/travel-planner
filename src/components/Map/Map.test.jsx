/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Map from './Map';

describe('Map', () => {
  it('renders Map without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Map center={[]} onError={() => {}} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Map snapshot', () => {
    const component = renderer.create(
      <Map center={[]} onError={() => {}} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
