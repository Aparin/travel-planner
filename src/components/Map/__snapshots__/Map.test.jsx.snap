// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map Map snapshot 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "height": "500px",
      "width": "800px",
    }
  }
/>
`;
